ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/components/DatePhraseTd.tsx

```diff
@@ -1,21 +1,26 @@
 import React from 'react';
 import { Tooltip } from '@patternfly/react-core';
 
-import { getDateTime, getDistanceStrictAsPhrase } from 'utils/dateUtils';
+import { getDateTime, getDistanceStrict, getDistanceStrictAsPhrase } from 'utils/dateUtils';
 
-export type DatePhraseTdProps = {
+export type DateDistanceTdProps = {
     date: string | number | Date | null | undefined;
+    asPhrase?: boolean;
 };
 
-function DatePhraseTd({ date }: DatePhraseTdProps) {
+function DateDistanceTd({ date, asPhrase = true }: DateDistanceTdProps) {
     if (!date) {
         return null;
     }
     return (
         <Tooltip content={getDateTime(date)}>
-            <span>{getDistanceStrictAsPhrase(date, new Date())}</span>
+            <span>
+                {asPhrase
+                    ? getDistanceStrictAsPhrase(date, new Date())
+                    : getDistanceStrict(date, new Date())}
```

Comment:
Although this seems to work here and elsewhere, I ask to increase my understanding of React: is `new Date()` a side-effect that bends the rules of pure rendering according to props?

Reply:
That's a good point. I'm not sure I would call it a side-effect since it doesn't mutate or do IO, but it does make a component impure in the sense that `props+state+context !== output`.

I suppose this could cause issues with re-rendering, especially if the `new Date()` was not passed to a leaf node. There is an argument to be made that the system time is an implicit input to a pure render method, but it is definitely questionable. It feels like `new Date()` should be passed to the top level render call and made available via context or something to be 100% pure. 😃